pilot/pkg/networking/core/configgen.go

```diff
@@ -23,6 +23,8 @@ import (
 )
 
 // ConfigGenerator represents the interfaces to be implemented by code that generates xDS responses
+// Deprecated: the interface is too coarse, when we have multiple generator implementations we can
+//  try to abstract (if needed), but right now it creates more impedance miss-match pain than benefits.
```

Comment:
No. the idea is to let others reuse this library in their implementations, rather then forking istio just to add custom filters or other customizations. We can certainly add more fine grained functions for inbound/outbound